parts/linux/cloud-init/artifacts/ensure-no-dup.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+# remove this if we are no longer using promiscuous bridge mode for containerd
+# background: we get duplicated packets from pod to serviceIP if both are on the same node (one from the cbr0 bridge and one from the pod ip itself via kernel due to promiscuous mode being on)
+# we should filter out the one from pod ip
+# this is exactly what kubelet does for dockershim+kubenet
+# https://github.com/kubernetes/kubernetes/pull/28717
+
+ebtables -t filter -L AKS-DEDUP-PROMISC 2>/dev/null
+if [[ $? -eq 0 ]]; then
+    echo "AKS-DEDUP-PROMISC rule already set"
+    exit 0
+fi
+if [[ ! -f /etc/cni/net.d/10-containerd-net.conflist ]]; then
+    echo "cni config not up yet...exiting early"
+    exit 1
+fi
+
+bridgeName=$(cat /etc/cni/net.d/10-containerd-net.conflist  | jq -r ".plugins[] | select(.type == \"bridge\") | .bridge")
+promiscMode=$(cat /etc/cni/net.d/10-containerd-net.conflist  | jq -r ".plugins[] | select(.type == \"bridge\") | .promiscMode")
+if [[ "${promiscMode}" != "true" ]]; then
+    echo "bridge ${bridgeName} not in promiscuous mode...exiting early"
+    exit 0
+fi
+
+podSubnetAddr=$(cat /etc/cni/net.d/10-containerd-net.conflist  | jq -r ".plugins[] | select(.type == \"bridge\") | .ipam.subnet")
+
+if [[ ! -f /sys/class/net/${bridgeName}/address ]]; then
```

Comment:
just curious - I assume this is the same as ip addr show ${bridgeName}, but is there any advantage of one over another?

Reply:
No real advantage other than no need for greps/seds or fancy args passed to `ip addr`. Following the everything-is-a-file-in-linux philosophy. 
I'll see if I can do the same for ip address.

Reply:
seems like for ip address is not as easy. i'll need to do some parsing of the /proc/net/* files.

Reply:
no need to spend extra time. I was just curious.